store/local_cache_supplier.go

```diff
@@ -15,25 +15,31 @@ const (
 	REACTION_CACHE_SIZE = 20000
 	REACTION_CACHE_SEC  = 1800 // 30 minutes
 
+	ROLE_CACHE_SIZE = 20000
+	ROLE_CACHE_SEC  = 1800 // 30 minutes
```

Comment:
You could also do `(time.Minute * 30).Seconds()` and then the comment isn't required and it can never get out of sync with the code.

Reply:
Unfortunately, you can't do that in a `const` block.

Reply:
You could do it... You just couldn't use the `Seconds()` method:

```go
int(time.Minute * 30 / time.Second)
```

But that has the potential downside of being typed instead of untyped. My personal preference and recommendation:

```go
ROLE_CACHE_SEC = 30 * 60
```